rust/kernel/sync/spinlock.rs

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! A kernel spinlock.
+//!
+//! This module allows Rust code to use the kernel's [`struct spinlock`].
```

Comment:
I have to see how to handle `Doc/` links in the best way possible, but it would be nice to add e.g.:

```suggestion
//! This module allows Rust code to use the kernel's [`struct spinlock`].
//!
//! See https://www.kernel.org/doc/Documentation/locking/spinlocks.txt.
```

Reply:
Added new link.